src/main/java/org/ballerinax/datamapper/DataMapperNodeVisitor.java

```diff
@@ -0,0 +1,208 @@
+                    Collection<MethodSymbol> methods = ((BallerinaClassSymbol) symbol).methods().values();
+                    Map<String, FunctionRecord> functionMap = new HashMap<>();
+                    for (MethodSymbol method : methods) {
+                        boolean checkForRemote = false;
+                        for (Qualifier qualifier : method.qualifiers()) {
+                            if (qualifier == Qualifier.REMOTE) {
+                                checkForRemote = true;
+                                break;
+                            }
+                        }
+                        if (checkForRemote) {
+                            if (method.getName().isEmpty()) {
+                                continue;
+                            }
+                            String functionName = method.getName().get();
+                            FunctionRecord functionRecord = new FunctionRecord();
+                            for (ParameterSymbol parameter : method.typeDescriptor().parameters()) {
+                                if (parameter.getName().isPresent()) {
+                                    String parameterName = parameter.getName().get();
+                                    String paraType = parameter.typeDescriptor().typeKind().toString();
+                                    functionRecord.addParameter(parameterName, paraType);
+                                }
+                            }
+                            TypeSymbol returnTypeSymbol = method.typeDescriptor().returnTypeDescriptor().get();
+                            String returnName = null;
+                            if (returnTypeSymbol.typeKind() == TypeDescKind.UNION) {
+                                List<TypeSymbol> returnList = ((UnionTypeSymbol) returnTypeSymbol).
+                                        memberTypeDescriptors();
+                                for (TypeSymbol typeSymbol : returnList) {
+                                    if (typeSymbol.typeKind() == TypeDescKind.ERROR) {
+                                        continue;
+                                    } else {
+                                        returnName = typeSymbol.signature();
+                                        break;
+                                    }
+                                }
+                            } else {
+                                returnName = returnTypeSymbol.signature();
+                            }
+                            functionRecord.setReturnType(returnName);
+                            functionMap.put(functionName, functionRecord);
+                            checkForRemote = false;
+                        }
+                    }
+                    if (!functionMap.isEmpty()) {
+                        this.clientMap.put(clientName, functionMap);
+                    }
+                }
+            }
+        }
+    }
+}
+
+class FunctionRecord {
```

Comment:
Shall we move this to its own source file?

Reply:
Moved to a separate source file.